injection/sharedmain/main.go

```diff
@@ -249,6 +268,14 @@ func MainWithConfig(ctx context.Context, component string, cfg *rest.Config, cto
 			leaderElectionConfig.GetComponentConfig(component))
 	}
 
+	observabilityConfig, err := GetObservabilityConfig(ctx)
+	if err != nil {
+		logger.Fatal("Error loading observability configuration: ", err)
+	}
+	observabilityConfigMap := observabilityConfig.GetConfigMap()
+	metrics.ConfigMapWatcher(ctx, component, SecretFetcher(ctx), logger)(&observabilityConfigMap)
+	profilingHandler.UpdateFromConfigMap(&observabilityConfigMap)
```

Comment:
For parity with logging, I'd probably move this into a function like this one: https://github.com/knative/pkg/blob/714b7630a836ef3952826849f31fe9bc16493218/injection/sharedmain/main.go#L311-L313

Reply:
done, added function for setting up observability